x/ssi/types/utils.go

```diff
@@ -46,3 +47,84 @@ func FindInSlice(list []string, element string) bool {
 	}
 	return false
 }
+
+// checkDuplicateItems return a duplicate Id from the list, if found
+func checkDuplicateItems(list []string) string {
+	presentMap := map[string]bool{}
+	for idx := range list {
+		if _, present := presentMap[list[idx]]; !present {
+			presentMap[list[idx]] = true
+		} else {
+			return list[idx]
+		}
+	}
+	return ""
+}
+
+func getDocumentChainNamespace(docId string) string {
+	docIdElements := strings.Split(docId, ":")
+
+	// Non Blockchain Account ID MSI
+	if len(docIdElements) == 4 || len(docIdElements) == 6 {
+		return docIdElements[2]
+	} else {
+		return ""
+	}
+}
+
+func getDocumentIdentifier(docId string) (string, error) {
+	docIdElements := strings.Split(docId, ":")
+
+	if len(docIdElements) < 1 {
+		return "", fmt.Errorf("invalid document Id: %v", docId)
+	}
+
+	return docIdElements[0], nil
+}
+
+func getDocumentMethod(docId string) (string, error) {
+	docIdElements := strings.Split(docId, ":")
+
+	if len(docIdElements) < 2 {
+		return "", fmt.Errorf("invalid document Id: %v", docId)
+	}
+
+	return docIdElements[1], nil
+}
+
+// GetMethodSpecificIdAndType extracts the method-specific-id from Document Id and which of following
+// types it belongs to:
+//
+// 1. CAIP-10 Blockchain Account ID
+//
+// 2. String consisting of Alphanumeric, dot (.) and hyphen (-) characters only
+func GetMethodSpecificIdAndType(didId string) (string, string, error) {
+	docIdElements := strings.Split(didId, ":")
+	var methodSpecificId string
+	var methodSpecificIdCondition string
+
+	if getMSIBlockchainAccountIdCondition(didId) {
+		methodSpecificId = strings.Join(docIdElements[(len(docIdElements)-3):], ":")
+		methodSpecificIdCondition = MSIBlockchainAccountId
+	} else if getMSINonBlockchainAccountIdCondition(didId) {
+		methodSpecificId = docIdElements[len(docIdElements)-1]
+		methodSpecificIdCondition = MSINonBlockchainAccountId
+	} else {
+		return "", "", fmt.Errorf(
+			"unable to retrieve method-specific-id from DID Id: %v. It should either be a CAIP-10 Blockchain Account ID or a string consisting of alphanumeric, dot(.) and hyphen(-) characters only", didId)
+	}
+
+	return methodSpecificId, methodSpecificIdCondition, nil
+}
+
+// getMSINonBlockchainAccountIdCondition asserts if the Method Specific Id is a CAIP-10 Blockchain Account Id
+func getMSINonBlockchainAccountIdCondition(didId string) bool {
+	didIdElements := strings.Split(didId, ":")
+	return (len(didIdElements) == 3 || len(didIdElements) == 4)
+}
+
+// getMSIMultibaseCondition asserts if the Method Specific Id is a Multibase encoded string
+func getMSIBlockchainAccountIdCondition(didId string) bool {
```

Comment:
better change the variable name to reflect its a condition

Reply:
@Vishwas1  Renamed the following:

- `getMSIBlockchainAccountIdCondition` to `isMSIBlockchainAccountId`
- `getMSINonBlockchainAccountIdCondition` to `isMSINonBlockchainAccountId`